src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.Unix.cs

```diff
@@ -27,18 +27,26 @@ public NamedPipeTest_CurrentUserOnly_Unix(ITestOutputHelper output)
 
         [ConditionalTheory(nameof(IsRemoteExecutorSupportedAndOnUnixAndSuperUser))]
         [OuterLoop("Needs sudo access")]
-        [InlineData(PipeOptions.None, PipeOptions.None)]
-        [InlineData(PipeOptions.None, PipeOptions.CurrentUserOnly)]
-        [InlineData(PipeOptions.CurrentUserOnly, PipeOptions.None)]
-        [InlineData(PipeOptions.CurrentUserOnly, PipeOptions.CurrentUserOnly)]
+        [InlineData(PipeOptions.None, PipeOptions.None, PipeDirection.In)]
+        [InlineData(PipeOptions.None, PipeOptions.None, PipeDirection.InOut)]
+        [InlineData(PipeOptions.None, PipeOptions.CurrentUserOnly, PipeDirection.In)]
+        [InlineData(PipeOptions.None, PipeOptions.CurrentUserOnly, PipeDirection.InOut)]
+        [InlineData(PipeOptions.CurrentUserOnly, PipeOptions.None, PipeDirection.In)]
+        [InlineData(PipeOptions.CurrentUserOnly, PipeOptions.None, PipeDirection.InOut)]
+        [InlineData(PipeOptions.CurrentUserOnly, PipeOptions.CurrentUserOnly, PipeDirection.In)]
+        [InlineData(PipeOptions.CurrentUserOnly, PipeOptions.CurrentUserOnly, PipeDirection.InOut)]
         public async Task Connection_UnderDifferentUsers_BehavesAsExpected(
-            PipeOptions serverPipeOptions, PipeOptions clientPipeOptions)
+            PipeOptions serverPipeOptions, PipeOptions clientPipeOptions, PipeDirection clientPipeDirection)
         {
+            bool isRoot = AdminHelpers.IsProcessElevated();
```

Comment:
This test is executed only for "super user":

https://github.com/dotnet/runtime/blob/c1fab9cb40200269cde851d5dfc4763af8a27af4/src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.Unix.cs#L26-L28

https://github.com/dotnet/runtime/blob/d196d850079ba88e904123c725e0f358142349c7/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Unix.cs#L58

and the helper that checks whether the process is elevated performs same check:

https://github.com/dotnet/runtime/blob/d196d850079ba88e904123c725e0f358142349c7/src/libraries/Common/tests/TestUtilities/System/AdminHelpers.cs#L35-L40

It seems that this check is redundant and adding `PipeOptions.CurrentUserOnly` test cases has no effect?

Reply:
IIRC, the reason I did this was to complete the test matrix and have explicit behaviors of all combinations. But this could be represented by just having the theorydata where clientPipeOptions = CurrentUserOnly commented-out.

Reply:
@Jozkee IMO it would be better to remove the test cases that are always skipped and just add a comment to the test explaining why given scenarios are not covered.

Reply:
Actually, the right thing to do is remove `IsRemoteExecutorSupportedAndOnUnixAndSuperUser` that way we can run all the inline data when dotnet is _not_ running as sudo/root and skip the `clientPipeOptions == PipeOptions.CurrentUserOnly` when it is. 

This worked as I just described on my local wsl, I hope it works in CI as well.

Reply:
@Jozkee but we need to keep the `IsRemoteExecutorSupported` condition as the test is using it.